tensorboard/program.py

```diff
@@ -526,6 +526,14 @@ def get_url(self):
       host = self._flags.host
       display_host = (
           '[%s]' % host if ':' in host and not host.startswith('[') else host)
+
+    # Confirm that the connection is open, otherwise change to `localhost`
+    socket.setdefaulttimeout(1)
+    try:
+        socket.socket().connect((display_host, self.server_port))
+    except socket.timeout as e:
```

Comment:
Let's catch `socket.error` generally so that if there is some other error attempting the connection (not a timeout), we fall back to localhost in that case too.  For example, it looks like if the DNS name doesn't resolve at all, this line fails with `socket.gaierror` rather than `socket.timeout`:

```python
>>> socket.socket().connect(("foobartensorboard.com", 6006))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python2.7/socket.py", line 228, in meth
    return getattr(self._sock,name)(*args)
socket.gaierror: [Errno -2] Name or service not known
```

Reply:
Very well spotted, fixed in 71307.